tutorials/gke-less-disruptive-node-upgrades/index.md

```diff
@@ -0,0 +1,520 @@
+  provided by the surge node.
+
+## Before you begin
+
+We recommend that you use 
+[Cloud Shell](https://cloud.google.com/kubernetes-engine/docs/tutorials/hello-app#option_a_use_google_cloud_shell) 
+to run the commands in this tutorial. As an alternative, you can use your own workstation to run commands locally. in which
+case you would need to install the
+[required tools](https://cloud.google.com/kubernetes-engine/docs/tutorials/hello-app#option_b_use_command-line_tools_locally).
+
+This tutorial requires a Google Cloud project. You can use an existing project or 
+[create a new project](https://cloud.google.com/resource-manager/docs/creating-managing-projects#creating_a_project).
+
+This tutorial follows the
+[Deploying a containerized web application](https://cloud.google.com/kubernetes-engine/docs/tutorials/hello-app) tutorial.
+We recommended that you complete the previous tutorial before starting this one.
+
+If you didn't complete the
+[Deploying a containerized web application](https://cloud.google.com/kubernetes-engine/docs/tutorials/hello-app) 
+tutorial and would like to immediately start this tutorial, you can run the following commands to create a cluster with
+the `hello-app` application running on it:
+
+1.  Clone the repository and navigate to the app directory:
+
+        git clone https://github.com/GoogleCloudPlatform/kubernetes-engine-samples
+        cd kubernetes-engine-samples/hello-app
+
+1.  Set a variable for the
+    [project ID](https://cloud.google.com/resource-manager/docs/creating-managing-projects#identifying_projects)
+    by replacing `[PROJECT_ID]` with your project ID in this command:
+
+        export PROJECT_ID=[PROJECT_ID]
+        
+1.  Build and push the image:
+
+        docker build -t gcr.io/${PROJECT_ID}/hello-app:v1 .
+        gcloud auth configure-docker
+        docker push gcr.io/${PROJECT_ID}/hello-app:v1
+
+1.  Create a cluster
+
+        gcloud config set project $PROJECT_ID
+        gcloud config set compute/zone us-central1-a
+        gcloud container clusters create hello-cluster --machine-type=g1-small --num-nodes=3
+	
+1.  Deploy and expose the application:
+
+        gcloud container clusters get-credentials hello-cluster
+        kubectl create deployment hello-web --image=gcr.io/${PROJECT_ID}/hello-app:v1
+        kubectl expose deployment hello-web --type=LoadBalancer --port 80 --target-port 8080
+
+## Costs
+
+In this tutorial, you create a GKE cluster with 3 g1-small virtual machine instances. The total cost of the resources used
+during for this tutorial is estimated to be less than $0.10. For details, see
+[VM instances pricing](https://cloud.google.com/compute/vm-instance-pricing).
+
+## Modify hello-app to work with resources
+
+In this section, you modify the `hello-app` source code.
+
+If you don't want to edit the source code manually, you can download the updated version of `main.go` and overwrite your 
+local copy with it.
+
+```shell
```

Comment:
It would be best if the code were in the GoogleCloudPlatform repository, preferably together with the tutorial itself.

Reply:
The code _is_ in the same repo as the turorial.

Are you talking about the code downloaded with the below command?

curl https://raw.githubusercontent.com/tamasr/community/master/tutorials/gke-less-disruptive-node-upgrades/main.go -O

This just points to my repository and the URL should be updated to point to the new repo (probably as part of this review).

The new URL would look like

https://raw.githubusercontent.com/GoogleCloudPlatform/community/master/tutorials/gke-less-disruptive-node-upgrades/main.go

Do you want me to update them? Is there another (preferred) URL format?

Note: the idea here is that the user already cloned the repo used in "Deploying a containerized web application" (https://cloud.google.com/kubernetes-engine/docs/tutorials/hello-app) and modifies the code they have there.

Reply:
Got it.

Yes, please, go ahead and make that change.